src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(beast) {
+    if ((beast instanceof Herbivore) && (beast.hidden === false)) {
+      beast.health -= 50;
+      Animal.alive = Animal.alive.filter(organism => organism.health > 0);
```

Comment:
why organism, but not animal?

Reply:
did not want to repeat the names of variables and classes and invented new